### PR TITLE
Day 8 solution

### PR DESCRIPTION
**Pull request by bstekas**

### CODE REVIEW

File: 2020/Day8.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    for i in INPUT:
+        i = i.split()
+        output.append([i[0], int(i[1])])
+
+    return output
+
+
+def execute_program(code):
+    restart = False
+    visited = []
```
Comment (zaobasy) [inline 2020/Day8.py:32]: I think it's slightly more efficient to use a set here. Containment in a set is faster than a list I think but likely not a huge difference. I'm just all about them sets

File: 2020/Day8.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    accumulator = 0
+
+    while not restart:
+        action, num = code[i]
+        visited.append(i)
+        if action in ['nop', 'acc']:
+            if action == 'acc':
+                accumulator += num
+            i += 1
+            if (i in visited) or (i == len(code)):
```
Comment (zaobasy) [inline 2020/Day8.py:43]: You repeat this block twice (the `if (i in visited)...`), you could move this check to the end outside the other ifs (unindent line 47). Also can probably avoid the nested if's by just checking each action independently.

File: 2020/Day8.py (hunk 1 above)
Comment (bstekas) [reply 2020/Day8.py:32]: I know you love sets, but I wanted to preserve the order they were visited.

Comment (zaobasy) [reply 2020/Day8.py:32]: That's fair, my solution doesn't actually provide the path you took.

File: 2020/Day8.py (hunk 2 above)
Comment (bstekas) [reply 2020/Day8.py:43]: Yeah I don't remember why I thought this was necessary. I do know that at first I thought `nop` and `acc` had the same behavior which is why they're grouped together.

**Pull request merged**